pkg/resources/function.go

```diff
@@ -161,402 +168,663 @@ func Function() *schema.Resource {
 	}
 }
 
-// CreateFunction implements schema.CreateFunc.
-func CreateFunction(d *schema.ResourceData, meta interface{}) error {
+func CreateContextFunction(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	lang := strings.ToUpper(d.Get("language").(string))
+	switch lang {
+	case "JAVA":
+		return createJavaFunction(ctx, d, meta)
+	case "JAVASCRIPT":
+		return createJavascriptFunction(ctx, d, meta)
+	case "PYTHON":
+		return createPythonFunction(ctx, d, meta)
+	case "SCALA":
+		return createScalaFunction(ctx, d, meta)
+	case "": // SQL if language is not set
```

Comment:
add "scala" as one of the allowed values in the "language" global variable used in the ValidateFunc of "language" schema field, and add another case for "SQL". If user writes `language = "SQL"` then it will throw, because it will land into default statement. Make it `case "", "SQL":`

Reply:
added